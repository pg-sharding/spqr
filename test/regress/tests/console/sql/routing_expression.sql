
create distribution ds1 column types int hash;
create relation tr(MURMUR [id1 INT , id2 VARCHAR]);

CREATE KEY RANGE FROM 3221225472 ROUTE TO sh4;
CREATE KEY RANGE FROM 2147483648 ROUTE TO sh3;
CREATE KEY RANGE FROM 1073741824 ROUTE TO sh2;
CREATE KEY RANGE FROM 0 ROUTE TO sh1;

show relations;

DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
