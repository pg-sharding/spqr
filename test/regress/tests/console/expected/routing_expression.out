
		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

create distribution ds1 column types int hash;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

create relation tr(MURMUR [id1 INT HASH, id2 VARCHAR HASH]);
      attach table      
------------------------
 relation name   -> tr
 distribution id -> ds1
(2 rows)

CREATE KEY RANGE FROM 3221225472 ROUTE TO sh4;
    add key range    
---------------------
 bound -> 3221225472
(1 row)

CREATE KEY RANGE FROM 2147483648 ROUTE TO sh3;
    add key range    
---------------------
 bound -> 2147483648
(1 row)

CREATE KEY RANGE FROM 1073741824 ROUTE TO sh2;
    add key range    
---------------------
 bound -> 1073741824
(1 row)

CREATE KEY RANGE FROM 0 ROUTE TO sh1;
 add key range 
---------------
 bound -> 0
(1 row)

show relations;
 Relation name | Distribution ID | Distribution key | Schema name  
---------------+-----------------+------------------+--------------
 tr            | ds1             | ("", murmur)     | $search_path
(1 row)

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

